src/main/java/com/conveyal/datatools/manager/jobs/MergeFeedsJob.java

```diff
@@ -0,0 +1,584 @@
+        // If the merge type is regional, the file string should be equivalent to projectId, which is used by the client
+        // to download the merged feed upon job completion.
+        this.projectId = mergeType.equals(REGIONAL) ? file : null;
+        this.mergeType = mergeType;
+        // Assuming job is successful, mergedVersion will contain the resulting feed version.
+        this.mergedVersion = mergeType.equals(REGIONAL) ? null : new FeedVersion(this.feedSource);
+        this.mergeFeedsResult = new MergeFeedsResult(mergeType);
+    }
+
+    /**
+     * The final stage handles clean up (deleting temp file) and adding the next job to process the new merged version
+     * (assuming the merge did not fail).
+     */
+    public void jobFinished() {
+        // Delete temp file to ensure it does not cause storage bloat. Note: merged file has already been stored
+        // permanently.
+        if (!mergedTempFile.delete()) {
+            // FIXME: send to bugsnag?
+            LOG.error("Merged feed file {} not deleted. This may contribute to storage space shortages.", mergedTempFile.getAbsolutePath());
+        }
+    }
+
+    /**
+     * Primary job logic handles collecting and sorting versions, creating a merged table for all versions, and writing
+     * the resulting zip file to storage.
+     */
+    @Override
+    public void jobLogic() throws IOException {
+        // Create temp zip file to add merged feed content to.
+        mergedTempFile = File.createTempFile(filename, null);
+        mergedTempFile.deleteOnExit();
+        // Create the zipfile.
+        ZipOutputStream out = new ZipOutputStream(new FileOutputStream(mergedTempFile));
+        LOG.info("Created project merge file: " + mergedTempFile.getAbsolutePath());
+        List<FeedToMerge> feedsToMerge = collectAndSortFeeds(feedVersions);
+
+        // Determine which tables to merge (only merge GTFS+ tables for MTC extension).
+        final List<Table> tablesToMerge = Arrays.stream(Table.tablesInOrder)
+                                                .filter(Table::isSpecTable)
+                                                .collect(Collectors.toList());
+        if (DataManager.isExtensionEnabled("mtc")) {
+            // Merge GTFS+ tables only if MTC extension is enabled. We should do this for both regional and MTC merge
+            // strategies.
+            tablesToMerge.addAll(Arrays.asList(GtfsPlusTable.tables));
+        }
+        int numberOfTables = tablesToMerge.size();
+        // Loop over GTFS tables and merge each feed one table at a time.
+        for (int i = 0; i < numberOfTables; i++) {
+            Table table = tablesToMerge.get(i);
+            double percentComplete = Math.round((double) i / numberOfTables * 10000d) / 100d;
+            status.update( "Merging " + table.name, percentComplete);
+            // Perform the merge.
+            byte[] tableOut = constructMergedTable(table, feedsToMerge);
```

Comment:
Is there a way to use some kind of streaming file writer instead of exporting the whole output into a byte[]? That way, much less memory would be consumed and the writing can happen as the program moves along instead of all at once with the zip file writing.

Reply:
Yes, thanks for the suggestion. Changes inbound.